jxl_simd/src/aarch64/neon.rs

```diff
@@ -122,6 +122,140 @@ impl F32SimdVec for F32VecNeon {
         unsafe { vst1q_f32(mem.as_mut_ptr(), self.0) }
     }
 
+    #[inline(always)]
+    fn store_interleaved_2(a: Self, b: Self, base: &mut [f32], offset: usize) {
+        assert!(base.len() >= offset + 2 * Self::LEN);
+        // SAFETY: we just checked that `base` has enough space.
+        unsafe {
+            // vst2q_f32 stores two vectors interleaved
+            vst2q_f32(base.as_mut_ptr().add(offset), float32x4x2_t(a.0, b.0));
+        }
+    }
+
+    #[inline(always)]
+    fn store_interleaved_4(a: Self, b: Self, c: Self, d: Self, base: &mut [f32], offset: usize) {
+        assert!(base.len() >= offset + 4 * Self::LEN);
+        // SAFETY: we just checked that `base` has enough space.
+        unsafe {
+            // vst4q_f32 stores four vectors interleaved
+            vst4q_f32(
+                base.as_mut_ptr().add(offset),
+                float32x4x4_t(a.0, b.0, c.0, d.0),
+            );
+        }
+    }
+
+    #[inline(always)]
+    fn store_interleaved_8(
+        a: Self,
+        b: Self,
+        c: Self,
+        d: Self,
+        e: Self,
+        f: Self,
+        g: Self,
+        h: Self,
+        base: &mut [f32],
+        offset: usize,
+    ) {
+        assert!(base.len() >= offset + 8 * Self::LEN);
+
+        #[target_feature(enable = "neon")]
+        #[inline]
+        unsafe fn store_interleaved_8_impl(
```

Comment:
We don't need the `unsafe` here, as long as we move the checks on `base` inside the function (and thus pass in a slice and not a pointer).